### PR TITLE
Alovelltroy/quickstart

### PR DESCRIPTION
**Pull request by alexlovelltroy**
- Also added a docker-compose example to show how to dynamically add hosts without SIGHUP

- Readme describes basic usage

-No ipxe payload included

### CODE REVIEW

**Comment by njones-lanl** [comment]
I wonder if it's possible to either:
- compile ipxe as part of the build 
- ask the wolfi folks to include the ipxe rpm.

**Comment by njones-lanl** [comment]
Sounds like @travisbcotton is working on the ipxe compile now.

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+# Add the uefi ipxe binary to the tftp directory
+# TODO: Find a palce to download the latest version for inclusion
+
+VOLUME /etc/dnsmasq
+
+EXPOSE 53 53/udp
+EXPOSE 67 67/udp
+
+
+ENTRYPOINT ["dnsmasq", "-k", "--log-dhcp", "--log-facility=-", "-R", "-h", "-C", "/etc/dnsmasq/dnsmasq.conf" ]
```
Comment (njones-lanl) [inline Dockerfile:35]: Without a resource to rewrite the hosts file + sighup the process (from inside or outside the container) this seems like an incomplete solution IMO

Comment (alexlovelltroy) [reply Dockerfile:35]: This probably needs better documentation, but the dnsmasq.conf file refers to `dhcp-hostsdir`which is described in the man page as:

> --dhcp-hostsdir=<path>
>     This is equivalent to --dhcp-hostsfile, except for the following. The path MUST be a directory, and not an individual file. Changed or new files within the directory are read automatically, without the need to send SIGHUP. If a file is deleted or changed after it has been read by dnsmasq, then the host record it contained will remain until dnsmasq receives a SIGHUP, or is restarted; ie host records are only added dynamically. The order in which the files in a directory are read is not defined.  

This should allow us to add hosts dynamically without a SIGHUP.  Where we need a SIGHUP, we can handle that with the docker command `docker kill -s HUP <container>` will send a HUP to the main process of the container.  I'm comfortable with this as a manual process until we can replace dnsmasq with a dhcp server that operates using plugins.

**Comment by alexlovelltroy** [comment]
This PR needs a second container that periodically populates the dhcp configuration.

File: Dockerfile (hunk 1 above)
Comment (njones-lanl) [reply Dockerfile:35]: I think that still leaves us with a hole where people try to swap a blade or similar (ie create a host that replaces an old host).

**Comment by njones-lanl** [comment]
lgtm

**Pull request merged**